File: RealTimeTicketingSystem/RealimeTicketingSystem-backend/src/main/java/com/ticketing/RealimeTicketingSystem_backend/Models/TicketStatus.java
package com.ticketing.RealimeTicketingSystem_backend.Models;

import lombok.Data;

/**
 * TicketStatus class is used to store the status of the tickets
 */
@Data
public class TicketStatus {
    private int remainingTickets; // Tickets still available for customers.
    private int totalTicketsReleased; // Total tickets released so far.
    private boolean allTicketsReleased; // Total tickets released so far.
    private boolean allTicketsSold; // Whether all tickets have been sold.

    /**
     * Default constructor for TicketStatus.
     * Initializes the ticket status with default values.
     */
    public TicketStatus() {
        // Default constructor
    }
}

File: RealTimeTicketingSystem/RealTimeTicketingSystem-frontend/src/app/models/ticketStatus.ts
export interface TicketStatus {
    remainingTickets: number; // Number of remaining tickets that aren't sold yet
    totalTicketsReleased: number; // Total number of tickets released by vendors
    allTicketsReleased: boolean; // Indicating if all tickets have been released
    allTicketsSold: boolean; // Indicating if all tickets have been sold.
  }
  